### ISSUE TITLE
[MNG-3959] Per-execution inherited flag ignored

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#5801]
**[Dave Syer](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=david_syer)** opened **[MNG-3959](https://issues.apache.org/jira/browse/MNG-3959?redirect=false)** and commented

Per-execution inherited flag ignored.  E.g.

```
<plugin>
	<groupId>org.apache.maven.plugins</groupId>
	<artifactId>maven-antrun-plugin</artifactId>
	<executions>
		<execution>
			<id>test</id>
			<phase>initialize</phase>
			<inherited>false</inherited>
...
```

If you move the inherited element up to the plugin level it works, but then that applies to all executions, which isn't what is needed.

---

**Affects:** 2.2.1

2 votes, 3 watchers

**Comment by jira-importer** [comment]
**[Christian Schulte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=schulte77)** commented

Maven 3 correctly supports per-execution inherited flags.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Sadly, I can confirm this on Maven 2.2.1. Maven 3.0.x works as expected.

**Comment by jira-importer** [comment]
**[Jason van Zyl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jvanzyl)** commented

This works in 3.x, we're likely not going to fix it in 2.x.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Won't fix for Maven 2.2.x, works in Maven 3.x
